### PR TITLE
[META 388] Add support for Azure App Service Cloud metadata

### PR DESCRIPTION
**Pull request by russcam**
This commit adds support for retrieving cloud metadata about an Azure App Service from the environment variables.

### CODE REVIEW

File: src/Elastic.Apm/Helpers/EnvironmentHelper.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+			}
+			catch (Exception e)
+			{
+				logger.Debug()?.LogException(e,
+					"Error while getting environment variable {EnvironmentVariable}", key);
+			}
+
+			return null;
+		}
+
```
Comment (gregkalapos) [inline src/Elastic.Apm/Helpers/EnvironmentHelper.cs:31]:
```suggestion
```

nit: This seems to be unused - from what I see you read valued directly from the `IDictionary` returned from the method below. Which is fine I think.

**Review by gregkalapos — APPROVED** [review]
LGTM 👍 

1 small thing below.

File: src/Elastic.Apm/Helpers/EnvironmentHelper.cs (hunk 1 above)
Comment (russcam) [reply src/Elastic.Apm/Helpers/EnvironmentHelper.cs:31]: AIWYAGNI (Add It When You Are Gonna Need It) 😄

**Comment by russcam** [comment]
failing test is unrelated to this change:

https://apm-ci.elastic.co/job/apm-agent-dotnet/job/apm-agent-dotnet-mbp/job/PR-1083/5/testReport/Elastic.Apm.Tests.ApiTests/ConvenientApiTransactionTests/Initializing___Parallel___Windows__NET_Framework___Test___ConvenientApiTransactionTests_Elastic_Apm_Tests_ApiTests_ConvenientApiTransactionTests_AsyncTaskWithReturnType/

**Pull request merged**